hibernate-core/src/main/java/org/hibernate/type/descriptor/java/DurationJavaDescriptor.java

```diff
@@ -30,15 +44,22 @@ public String toString(Duration value) {
 		if ( value == null ) {
 			return null;
 		}
-		return String.valueOf( value.toNanos() );
+		return new StringBuilder()
```

Comment:
It seems we can specify the capacity of `StringBuilder` to avoid relocating new char array and copying (it would definitively happen for the internal char array's length is 16 by default).

Reply:
I don't think `toString()` is used for anything important, much less for anything performance sensitive, but there's a *much* worse problem right in on the next line that I just fixed, so I fixed your objection while I was at it.

Reply:
no objection at all. As a purely newbie, currently I can only find some minor programming imperfection. It is a great honour to review your code, :). Because of the same reason, I am afraid I am not fully qualified to approve it, unless I am pretty sure. Thanks for the quick response. It is exciting to interact with you here.